api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -207,10 +207,20 @@ func (mw *azureManagedControlPlaneWebhook) ValidateUpdate(ctx context.Context, o
 		allErrs = append(allErrs, err)
 	}
 
+	oldDNSPrefix := old.Spec.DNSPrefix
+	newDNSPrefix := m.Spec.DNSPrefix
+	if reflect.ValueOf(oldDNSPrefix).IsZero() {
+		oldDNSPrefix = ptr.To(old.Name)
+	}
+
+	if reflect.ValueOf(newDNSPrefix).IsZero() {
+		newDNSPrefix = ptr.To(m.Name)
+	}
```

Comment:
Can we safely assume that the new DNS prefix is already defaulted here since this same logic exists in the defaulting webhook?

Reply:
So, first I did kept the nil comparison but something broke in the unit tests, and the defaulter seems to not default correctly. I'll remove the reflect and the check inside webhook in order to see what unit tests are failing.